.changeset/introduce_includes.md

```diff
@@ -1,44 +1,66 @@
+`includes` uses a different glob pattern format that fixes [many](https://github.com/biomejs/biome/issues/2421) [issues](https://github.com/biomejs/biome/issues/3345) and limitations that Biome users reported.
+
+`includes` accepts an array of glob patterns.
+A glob pattern starting with a `!` is an exception.
+This replaces `ignore` patterns and allows users to create chains of include and ignore patterns.
+Thus, it is now possible to include again a file previously ignored.
+This was not possible with `include` and `ignore`, because `ignore` was always applied after `include`.
+
+`includes` handles `*` and `**/*` differently.
+The first pattern matches all files in the directory in which the Biome configuration file is.
+The second pattern matches all files.
+`include` and `ignore` interprets `*` as `**/*`.
+
+Let's take a Biome configuration with `include`/`ignore` and convert it into the new `includes` field.
+In the following configuration, all files of the `test` directory and all files ending with the extension `.test.js` are ignored.
+The linter is only executed on files if the `src` directory that doesn't end with the `.gen.js` extension.
+Also, an extra rule is enabled on files ending with the `.ts` extension.
 
 ```json
 {
     "files": {
+        "ignore": ["*.test.js", "test"]
+    },
+    "linter": {
         "include": ["src/**"],
-        "ignore": ["**/*.test.js"]
-    }
+        "ignore": ["*.gen.js"],
+        "enabled": true
+    },
+    "overrides": [{
+        "include": ["*.ts"],
+        "linter": { "rules": { "style": { "noDefaultExport": "on" } } }
+    }]
 }
 ```
 
-Some Biome users have requested the ability to ignore a set of files except some of the files.
-With the current system, this is not possible because `include` is always applied before `ignore`.
-
-Also, many Biome users [reported](https://github.com/biomejs/biome/issues/2421) [issues](https://github.com/biomejs/biome/issues/3345) with the behavior of the glob patterns.
-Notably:
-
-- `src/**` is interpreted as `**/src/**`
-- `*.js` is interpreted as `**/*.js`
-
-To solve all these issues, we introduce a new field `includes`, which replaces both `include` and `ignore`.
-`includes` accepts an array of glob patterns with a stricter and more intuitive behavior than the previous glob pattern format.
-A glob starting with a `!` is an exception.
-This replaces `ignore` patterns.
-
-The previous configuration must be updated as follows:
+The Biome configuration is migrated to the following one:
 
 ```json
 {
     "files": {
-        "includes": ["src/**", "!**/*.test.js"]
-    }
+        "includes": ["**", "!**/*.test.js", "!test"]
+    },
+    "linter": {
+        "includes": ["src/**", "!**/*.gen.js"],
+        "enabled": true
+    },
+    "overrides": [{
+        "includes": ["**/*.ts"],
+        "linter": { "rules": { "style": { "noDefaultExport": "on" } } }
+    }]
```

Comment:
Shouldn't we explain the new globs, like, in relation to the bugs that were mentioned? For example, we should explain that `*.ts` was migrated to `**/*.ts`.

Reply:
I improved the example. Let me know if it is better. :)